WordPress.gitignore

```diff
@@ -1,23 +1,34 @@
-# ignore everything in the root except the "wp-content" directory.
+# Wordpress - ignore core, configuration, examples, generated data and logs.
+# https://github.com/github/gitignore/blob/master/WordPress.gitignore
+
+# Core
+/*
 !wp-content/
 
-# ignore everything in the "wp-content" directory, except:
-# "mu-plugins", "plugins", "themes" directory
+# Configuration
+wp-config.php
+
+# Generated data without custom one
 wp-content/*
 !wp-content/mu-plugins/
 !wp-content/plugins/
+wp-content/plugins/index.php
 !wp-content/themes/
+wp-content/themes/index.php
 
-# ignore these plugins
+# Example plugins
 wp-content/plugins/hello.php
 
-# ignore specific themes
+# Example themes
 wp-content/themes/twenty*/
 
-# ignore node dependency directories
+# Example themes use node dependency
 node_modules/
 
-# ignore log files and databases
+# Log files and databases
 *.log
 *.sql
```

Comment:
Why are we ignoring `.sql` files? Do they naturally come/are generated on a typical WordPress installation? Aren't we over-assuming this?

Reply:
Wordpress require SQL database. I don't find that they pre-exist on first installation. 
Remove it? I don't see value to have it in this ignore file template

Reply:
Yes, I think the `*.sql` and `*.sqlite` rules should be removed. I know some people might have them as part of their workflows, but I see this gitignore repository more as **base/starter gitignore files**, that can be mixed together (like using [gitignore.io](http://gitignore.io))

I see these files only containing (1) the specific files for each technology/tool and (2) and _possibly_ rules for widely common configurations of such technologies/tools (ex: `*.log` line).

Also, they shouldn't have redundant lines that are covered by other specific gitignores (ex: [`Global/macOS.gitignore`](https://github.com/github/gitignore/blob/master/Global/macOS.gitignore)).

Project/workflow specific rules should be added/maintained individually on each project, IMHO.

Reply:
Btw, added my own contribution addressing the above and more on #3295.